mcumgr-core/src/main/java/io/runtime/mcumgr/util/CBOR.java

```diff
@@ -6,54 +6,153 @@
 
 package io.runtime.mcumgr.util;
 
-import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.type.CollectionType;
+import com.fasterxml.jackson.databind.type.MapType;
 import com.fasterxml.jackson.dataformat.cbor.CBORFactory;
 
-import java.io.ByteArrayInputStream;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
 import java.io.IOException;
-import java.util.HashMap;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
 import java.util.Map;
 
+
 @SuppressWarnings("unused")
 public class CBOR {
-    private final static CBORFactory sFactory = new CBORFactory();
 
-    public static byte[] toBytes(Object obj) throws IOException {
-        ObjectMapper mapper = new ObjectMapper(sFactory);
-        return mapper.writeValueAsBytes(obj);
-    }
+    private final static ObjectMapper MAPPER = new ObjectMapper(new CBORFactory());
+
+    //*********************************************************************************************
+    // Encode
+    //*********************************************************************************************
 
-    public static <T> T toObject(byte[] data, Class<T> type) throws IOException {
-        ObjectMapper mapper = new ObjectMapper(sFactory);
-        return mapper.readValue(data, type);
+    /*
+     * Bytes
+     */
```

Comment:
Not sure what this comment is supposed to convey (that the method name doesn't)?